### DISCUSSION TITLE
Federated exchanges - max-hops=2 duplicate messages (full graph)

### DISCUSSION THREAD

**Discussion by kotyara85** [rabbitmq/rabbitmq-server#4693] [Questions] [answered]
Hello, 
We have a full graph setup 
       Cluster1 
 /                     \
Cluster2 --- Cluster 3

with max-hop=2, this's in order if one of the links if broken and we still want to make sure the message are getting delivered
In case if I publish a message to a routing key which is present on Cluster3 from Cluster1, it will get delivered twice
Is there any way to prevent rmq to transmit the message twice, let's say by choosing the shortest path?
Thanks

Thanks

**Comment by michaelklishin** [comment, marked as answer]
Exchange federation *replicates*, and does not deduplicate. There is a queue-level deduplication plugin maintained by a community member, or your consumers can do it using a key/value store or a service such as Consul, etcd.

**Reply by kotyara85** [reply]
Makes sense, yeah I was thinking about k-v store to track messages. Thanks
